src/mb_enhanced_cover_art_uploads/fetch.ts

```diff
@@ -140,14 +143,31 @@ export class ImageFetcher {
             }
         }
 
-        this.#doneImages.add(url.href);
+        if (!hasMoreImages && !onlyFront) {
+            // Don't mark the whole provider URL as done if we haven't grabbed
+            // all images.
+            this.#doneImages.add(url.href);
+        }
+
+        if (hasMoreImages) {
+            LOGGER.warn(`Not all images were fetched: ${images.length - finalImages.length} covers were skipped.`);
+        }
```

Comment:
```suggestion
        if (!hasMoreImages) {
            // Don't mark the whole provider URL as done if we haven't grabbed
            // all images.
            this.#doneImages.add(url.href);
        } else {
            LOGGER.warn(`Not all images were fetched: ${images.length - finalImages.length} covers were skipped.`);
        }
```
If we added only the fornt and there are noe more images we can still mark the whole provider URL as done.

Reply:
Seems fair. I originally split them up because Bandcamp/Soundcloud won't return track images if we're grabbing only the front cover, but realistically, if someone's using this option, they probably won't go back to grab the track images anyway.